lib/kasket/read_mixin.rb

```diff
@@ -26,7 +26,7 @@ def find_by_sql_with_kasket(*args)
 
       if query && has_kasket_index_on?(query[:index])
         if query[:key].is_a?(Array)
-          find_by_sql_with_kasket_on_id_array(query[:key])
+          filter_pending_records(find_by_sql_with_kasket_on_id_array(query[:key]))
```

Comment:
This is the fix. This should have been added in [the old PR](https://github.com/zendesk/kasket/pull/31/files#diff-127af90f8c53e8b5ca83aa0b9b0907cd57bfba8243d8e551e14e7a9b0dfebf6b), but it wasn't.